src/uu/env/src/split_iterator.rs

```diff
@@ -0,0 +1,505 @@
+                        // So apparently only known sequences are allowed, even though they are not expanded.... bug of GNU?
+                        self.push_ascii_char_to_word(BACKSLASH)?;
+                        self.take_one()?;
+                        SingleQuoted
+                    }
+                    Some(c) => return Err(self.make_invalid_sequence_backslash_xin_minus_s(c)),
+                },
+                DoubleQuoted => match c {
+                    None => {
+                        return Err(ParseError::MissingClosingQuote {
+                            pos: self.get_parser().get_look_at_pos(),
+                            c: '"',
+                        })
+                    }
+                    Some(b'$') => {
+                        self.substitute_variable()?;
+                        self.state
+                    }
+                    Some(DOUBLE_QUOTES) => {
+                        self.skip_one()?;
+                        Unquoted
+                    }
+                    Some(BACKSLASH) => {
+                        self.skip_one()?;
+                        DoubleQuotedBackslash
+                    }
+                    Some(_) => {
+                        self.take_one()?;
+                        DoubleQuoted
+                    }
+                },
+                DoubleQuotedBackslash => match c {
+                    None => {
+                        return Err(ParseError::MissingClosingQuote {
+                            pos: self.get_parser().get_look_at_pos(),
+                            c: '"',
+                        })
+                    }
+                    Some(b'\n') => {
+                        self.skip_one()?;
+                        DoubleQuoted
+                    }
+                    Some(DOUBLE_QUOTES) | Some(b'$') | Some(BACKSLASH) => {
+                        self.take_one()?;
+                        DoubleQuoted
+                    }
+                    Some(b'c') => {
+                        return Err(ParseError::BackslashCNotAllowedInDoubleQuotes {
+                            pos: self.get_parser().get_look_at_pos(),
+                        })
+                    }
+                    Some(c) if self.check_and_replace_ascii_escape_code(c)? => DoubleQuoted,
+                    Some(c) => return Err(self.make_invalid_sequence_backslash_xin_minus_s(c)),
+                },
+                Comment => match c {
+                    None => break,
+                    Some(b'\n') => {
+                        self.skip_one()?;
+                        Delimiter
+                    }
+                    Some(_) => {
+                        self.get_parser_mut().skip_until_ascii_char_or_end(b'\n')?;
+                        Comment
+                    }
+                },
+            };
+
+            if c.is_none() {
+                break;
+            }
+        }
+
+        Ok(mem::take(&mut self.words))
```

Comment:
If `split` takes `self` instead of `&mut self` we can just return `self.words` here. This also prevents us from ever reusing a parser (which would be in an invalid state).

Reply:
Thats a neat trick. Thanks. I was wondering how consuming self in a member function would ever be usefull. Here we have a usefull case. :-)